kong/plugins/http-log/handler.lua

```diff
@@ -72,37 +48,45 @@ local function log(premature, conf, body, name)
   end
   name = "[" .. name .. "] "
 
-  local ok, err
+  local headers
+  local req_url
   local parsed_url = parse_url(conf.http_endpoint)
-  local host = parsed_url.host
-  local port = tonumber(parsed_url.port)
-
-  local sock = ngx.socket.tcp()
-  sock:settimeout(conf.timeout)
 
-  ok, err = sock:connect(host, port)
-  if not ok then
-    ngx.log(ngx.ERR, name .. "failed to connect to " .. host .. ":" .. tostring(port) .. ": ", err)
-    return
+  if parsed_url.userinfo then
+    -- for lua-resty-http client Authorization
+    headers = {
+      ["Content-Type"] = conf.content_type,
+      ["Authorization"] = string_format(
+        "Basic %s", ngx.encode_base64(parsed_url.userinfo))
+    }
+    req_url = parsed_url.scheme .. "://" .. parsed_url.host .. ":" .. parsed_url.port
+      .. parsed_url.path .. (parsed_url.params or "")
+  else
+    headers = {
+      ["Content-Type"] = conf.content_type
+    }
+    req_url = conf.http_endpoint
   end
 
-  if parsed_url.scheme == HTTPS then
-    local _, err = sock:sslhandshake(true, host, false)
-    if err then
-      ngx.log(ngx.ERR, name .. "failed to do SSL handshake with " .. host .. ":" .. tostring(port) .. ": ", err)
-    end
+  local httpc = http.new()
+  httpc:set_timeout(conf.timeout)
+  local res, err = httpc:request_uri(req_url, {
+      ssl_verify = false,
+      method = conf.method:upper(),
+      body = body,
+      headers = headers
+  })
+
+  if not res then
+    ngx.log(ngx.ERR, name .. "request failed: " .. req_url, err)
+    return
   end
 
-  ok, err = sock:send(generate_post_payload(conf.method, conf.content_type, parsed_url, body))
-  if not ok then
-    ngx.log(ngx.ERR, name .. "failed to send data to " .. host .. ":" .. tostring(port) .. ": ", err)
+  if res.status >= 400 then
+    ngx.log(ngx.ERR, name .. "request failed: " .. req_url .. "status: " .. tostring(res.status), err)
   end
 
-  ok, err = sock:setkeepalive(conf.keepalive)
-  if not ok then
-    ngx.log(ngx.ERR, name .. "failed to keepalive to " .. host .. ":" .. tostring(port) .. ": ", err)
-    return
-  end
+  httpc:set_keepalive(conf.timeout)
```

Comment:
I noticed this replaced the use of `conf.keepalive` with `conf.timeout` -- was this intentional? Also, error checking here was removed. `lua-resty-http` will return non-error both when the connection is kept alive, or when it is properly closed.

Reply:
It's my mistake to change `conf.keepalive` to `conf.timeout`. 

The test case of this plugin was disabled (It looks like the mockbin has an exception) so I not sure this change can pass the test. But verified in our production environment.